bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -0,0 +1,547 @@
+        channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_VEHICLE_CAPACITY);
+        updateState(channel, new QuantityType<>(vehicleCapacity, Units.WATT_HOUR));
+        double vehicleOdometer = loadpoint.getVehicleOdometer();
+        channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_VEHICLE_ODOMETER);
+        updateState(channel, new QuantityType<>(vehicleOdometer, MetricPrefix.KILO(SIUnits.METRE)));
+        boolean vehiclePresent = loadpoint.getVehiclePresent();
+        channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_VEHICLE_PRESENT);
+        if (vehiclePresent == true) {
+            updateState(channel, OnOffType.ON);
+        } else {
+            updateState(channel, OnOffType.OFF);
+        }
+        long vehicleRange = loadpoint.getVehicleRange();
+        channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_VEHICLE_RANGE);
+        updateState(channel, new QuantityType<>(vehicleRange, MetricPrefix.KILO(SIUnits.METRE)));
+        int vehicleSoC = loadpoint.getVehicleSoC();
+        channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_VEHICLE_SOC);
+        updateState(channel, new QuantityType<>(vehicleSoC, Units.PERCENT));
+        String vehicleTitle = loadpoint.getVehicleTitle();
+        channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_VEHICLE_TITLE);
+        updateState(channel, new StringType(vehicleTitle));
+    }
+
+    private void createChannel(String channel, String channelGroupId, ChannelTypeUID channelTypeUID, String itemType) {
+        ChannelUID channelToCheck = new ChannelUID(thing.getUID(), channelGroupId, channel);
+        if (thing.getChannel(channelToCheck) == null) {
+            ThingBuilder thingBuilder = editThing();
+            Channel testchannel = ChannelBuilder
+                    .create(new ChannelUID(getThing().getUID(), channelGroupId, channel), itemType)
+                    .withType(channelTypeUID).build();
+            thingBuilder.withChannel(testchannel);
+            updateThing(thingBuilder.build());
+        }
+    }
+
+    /**
+     * Make a HTTP request.
+     * 
+     * @param description request description for logger
+     * @param url full request URL
+     * @param method reguest method, e.g. GET, POST
+     * @return the response body or response_code 999 if request faild
+     */
+    private String httpRequest(@Nullable String description, String url, String method) {
+        String response = "";
+        try {
+            response = HttpUtil.executeUrl(method, url, LONG_CONNECTION_TIMEOUT_MILLISEC);
+            logger.trace("{} - {}, {} - {}", description, url, method, response);
+            return response;
+        } catch (IOException e) {
+            logger.warn("IO Exception - {} - {}, {} - {}", description, url, method, e.toString());
```

Comment:
This should probably be reduced to **debug** level - see https://www.openhab.org/docs/developer/guidelines.html#f-logging.